Can no longer compile without TLS

Rust 1.57, 2022

   Compiling amiquip v0.4.1
error[E0425]: cannot find value `TlsFeatureNotEnabled` in this scope
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/amiquip-0.4.1/src/connection.rs:469:9
    |
469 |         TlsFeatureNotEnabled.fail()
    |         ^^^^^^^^^^^^^^^^^^^^
    |
   ::: /root/.cargo/registry/src/github.com-1ecc6299db9ec823/amiquip-0.4.1/src/errors.rs:10:10
    |
10  | #[derive(Snafu, Debug)]
    |          ----- similarly named unit struct `TlsFeatureNotEnabledSnafu` defined here
    |
help: a unit struct with a similar name exists
    |
469 |         TlsFeatureNotEnabledSnafu.fail()
    |         ~~~~~~~~~~~~~~~~~~~~~~~~~
help: consider importing this unit variant
    |
416 |     use crate::Error::TlsFeatureNotEnabled;
    |

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -20,5 +20,5 @@ jobs:
       run: cargo build --verbose
     - name: Run tests
       run: |
-        cargo test --verbose
+        cargo test --no-default-features --verbose
         cargo test --features native-tls --verbose

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "amiquip"
-version = "0.4.1"
+version = "0.4.2"
 authors = ["John Gallagher <johnkgallagher@gmail.com>"]
 edition = "2018"
 build = "build.rs"

--- a/Changelog.md
+++ b/Changelog.md
@@ -1,3 +1,8 @@
+# Version 0.4.2 (2022-01-12)
+
+* Fix compilation error with default features disabled (#36)
+  * Add --no-default-features to CI
+
 # Version 0.4.1 (2022-01-04)
 
 * Upgrade several dependencies:

--- a/README.md
+++ b/README.md
@@ -20,7 +20,7 @@ For usage, see the [documentation](https://docs.rs/amiquip/) and
 
 ## Minimum Support Rust Version
 
-The minimum supported Rust version for amiquip 0.4.1 is currently Rust 1.46.0,
+The minimum supported Rust version for amiquip 0.4.2 is currently Rust 1.46.0,
 but that may change with a patch release (and could change with a patch release
 to a dependency without our knowledge).
 

--- a/src/connection.rs
+++ b/src/connection.rs
@@ -466,7 +466,7 @@ mod amqp_url {
 
     #[cfg(not(feature = "native-tls"))]
     fn open_amqps(_: Url, _: ConnectionOptions<Auth>, _: ConnectionTuning) -> Result<Connection> {
-        TlsFeatureNotEnabled.fail()
+        TlsFeatureNotEnabledSnafu.fail()
     }
 
     #[cfg(feature = "native-tls")]

